src/aleph/vm/pool.py

```diff
@@ -213,3 +213,17 @@ def get_instance_executions(self) -> Iterable[VmExecution]:
             for _vm_hash, execution in self.executions.items()
             if execution.is_running and execution.is_instance
         )
+
+    def get_executions_by_sender(self, payment_type: PaymentType) -> Iterable[Tuple[str, list[VmExecution]]]:
+        """Return all executions of the given type, grouped by sender and by chain."""
+        executions_by_sender: dict[str, list[VmExecution]] = {}
+        for vm_hash, execution in self.executions.items():
```

Comment:
I think that this should only get the persistent VMs for PAYG.

Reply:
Why? We can just calculate 1/10th of the price for on-demand VMs with PAYG, at least for now.

Later on we really need more accurate measurements and stream stop/start when using on-demand, or some different kind of payment method alltogether, if it doesn't prove feasible to rapidly start and stop streams.